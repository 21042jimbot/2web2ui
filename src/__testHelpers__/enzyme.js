export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
