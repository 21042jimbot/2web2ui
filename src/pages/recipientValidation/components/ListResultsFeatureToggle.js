import React from 'react';
import { connect } from 'react-redux';
import { isAccountUiOptionSet } from 'src/helpers/conditions/account';
import { selectCondition } from 'src/selectors/accessConditionState';
import JobsTableCollection from './JobsTableCollection';
import ListResults from './ListResults';

// TODO: Remove in SE-156
export const ListResultsFeatureToggle = ({ isFlagEnabled }) => {
  if (isFlagEnabled) {
    return <JobsTableCollection />;
  }

  return <ListResults />;
};


const mapStateToProps = (state) => ({
  isFlagEnabled: selectCondition(isAccountUiOptionSet('recipientValidationV2'))(state)
});

export default connect(mapStateToProps)(ListResultsFeatureToggle);
