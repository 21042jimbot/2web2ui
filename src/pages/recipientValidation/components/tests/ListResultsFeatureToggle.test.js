import React from 'react';
import { shallow } from 'enzyme';
import { ListResultsFeatureToggle } from '../ListResultsFeatureToggle';

describe('ListResultsFeatureToggle', () => {
  const subject = (props = {}) => shallow(
    <ListResultsFeatureToggle
      {...props}
    />
  );

  it('renders list results by default', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('renders jobs table collection', () => {
    expect(subject({ isFlagEnabled: true })).toMatchSnapshot();
  });
});
