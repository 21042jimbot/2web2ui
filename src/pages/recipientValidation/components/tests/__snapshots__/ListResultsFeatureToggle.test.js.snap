// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListResultsFeatureToggle renders jobs table collection 1`] = `<Connect(withJobs(JobsTableCollection)) />`;

exports[`ListResultsFeatureToggle renders list results by default 1`] = `<Connect(WithContext(ListResults)) />`;
